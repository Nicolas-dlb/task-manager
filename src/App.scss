.app {
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	text-align: center;
	height: 100vh;
	background: var(--main-bg-color);
}
