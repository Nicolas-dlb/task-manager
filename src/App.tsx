import { useContext, useEffect } from "react";
import { Navigate, Route, Routes } from "react-router-dom";
import { ThemeContext } from "./utils/providers/ThemeProvider";
import Dashboard from "./interfaces/Dashboard/Dashboard";
import { setBoards } from "./redux/reducers/boardsSlice";
import { useAppDispatch } from "./redux/hooks";
import Login from "./interfaces/Auth/Login/Login";
import Register from "./interfaces/Auth/Register/Register";
import { onAuthStateChanged } from "firebase/auth";
import { auth, db } from "./firebaseConfig";
import { doc, onSnapshot } from "firebase/firestore";
import { hideUselessLogs } from "./utils/helpers";
import LowerCaseRedirect from "./components/LowerCaseRedirect/LowerCaseRedirect";
import "./App.scss";

function App(): JSX.Element {
	const { theme } = useContext(ThemeContext);
	const dispatch = useAppDispatch();

	useEffect(() => {
		hideUselessLogs();
	}, []);

	onAuthStateChanged(auth, (user) => {
		if (user) {
			const userDoc = doc(db, `users/${user.uid}`);
			onSnapshot(userDoc, (snap) => {
				dispatch(setBoards(snap.data()?.boards));
			});
		}
	});

	return (
		<div className="app" data-theme={theme}>
			<LowerCaseRedirect />
			<Routes>
				<Route path="/auth">
					<Route path="login" element={<Login />} />
					<Route path="register" element={<Register />} />
				</Route>
				<Route path="/dashboard" element={<Dashboard />} />
				<Route path="*" element={<Navigate to="/auth/login" />} />
			</Routes>
		</div>
	);
}

export default App;
