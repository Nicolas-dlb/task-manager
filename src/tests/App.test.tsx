import { describe, it } from "vitest";
import App from "../App";
import renderWithStore from "./utils/renderWithStore";
import { BrowserRouter } from "react-router-dom";

describe("App", () => {
	it("Renders app properly", () => {
		const view = renderWithStore(
			<BrowserRouter>
				<App />
			</BrowserRouter>
		);

		expect(view).toMatchSnapshot();
	});
});
