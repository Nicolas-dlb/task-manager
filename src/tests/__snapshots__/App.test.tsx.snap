// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > Renders app properly 1`] = `undefined`;
